datafusion/physical-plan/src/joins/cross_join.rs

```diff
@@ -51,7 +51,7 @@ type JoinLeftData = (RecordBatch, MemoryReservation);
 
 /// executes partitions in parallel and combines them into a set of
 /// partitions by combining all values from the left with all values on the right
-#[derive(Debug)]
+#[derive(Debug)] // note not Clone because of the OnceAsync
```

Comment:
```suggestion
/// Clone trait not derived because the structure used with `OnceAsync` which doesn't expect the structure to be cloned
#[derive(Debug)]
```

Reply:
perhaps something like that?